using_images/other_images/jenkins.adoc

```diff
@@ -117,6 +117,87 @@ define a persistent volume pool in the OpenShift deployment.
 
 Once selected, you must link:../../dev_guide/templates.html#creating-resources-from-a-template[instantiate] the template to be able to use Jenkins.
 
+== Customizing Jenkins image
+
+=== Using Jenkins image as Source-To-Image builder
+
+The Source-To-Image allows you to do additional modifications to the Jenkins
+image. For example, you can use
+link:../../architecture/core_concepts/builds_and_image_streams.html#source-build[Source-To-Image]
+to copy your Jenkins jobs definitions, additional plugins or tweak the default
+configuration if you want.
+
+To do that, you can define a build configuration, that uses the Jenkins image as
+a builder image:
+
+[source,yaml]
+----
+kind: "BuildConfig"
+apiVersion: "v1"
+metadata:
+  name: "custom-jenkins-build" <1>
+spec:
+  triggers: <2>
+    -
+      type: "ImageChange"
+  source: <3>
+    type: "Git"
+    git:
+      uri: "git://github.com/<org>/<repository-name>.git"
+  strategy: <4>
+    type: "Source"
+    sourceStrategy:
+      from:
+        kind: "ImageStreamTag"
+        name: "jenkins:latest"
+  output: <5>
+    to:
+      kind: "ImageStreamTag"
+      name: "custom-jenkins:latest"
+----
+
+<1> This specification will create a new `*BuildConfig*` named *custom-jenkins-build*.
+<2> The ImageChange trigger will rebuild the custom Jenkins image every time when the official Jenkins image is updated.
+<3> The source repository that contains custom Jenkins configuration, plugins, jobs, etc.
+<4> The "Source" strategy uses the official Jenkins image as the builder.
+
+The source repository must have the following layout:
+
+* `./plugins` - directory that contains custom binary Jenkins plugins you want to install.
+* `./plugins.txt` - file that lists additional plugins you want to install in the `plugin:version` format.
+* `./configuration/jobs` - directory that contains user-defined Jenkins job definitions.
+* `./configuration/config.xml` - file that contains your custom Jenkins configuration.
+
+Note that the content of the `./configuration` directory will be copied into the
+`/var/lib/jenkins` directory, so you can include any additional configuration
+file there.
```

Comment:
not clear what you're suggesting here.  How would they include additional config in /var/lib/jenkins?  they'd have to bake it into the image, and even then it would be pointless since we mount a volume there, so we'd shadow anything they put there.